### ISSUE TITLE
phantomjs doesn't exit for some time after casperjs tests are done.

### ISSUE THREAD

**Issue by jsmirnov** [ariya/phantomjs#13904]
Hi!
I am having an issue with testing with casperjs + phantomjs
Basically the problem is that even after all tests were done by casperjs, it takes some time(several minutes) to return to shell.
`wernight/phantomjs` is basically current master of phantomjs.
at 1.9.7 there was a bunch of meesages coming from phantomjs saying about accessing remote urls.
I am using OSX, and VM where docker is running, starts using 100% CPU.
Might someone know the reasoning for this one?

I am using docker to test:

```
FROM wernight/phantomjs

USER root

RUN apt-get update && apt-get upgrade -y && apt-get install -y --force-yes --no-install-recommends git-core libfreetype6 libfontconfig python
WORKDIR /srv
RUN git clone https://github.com/n1k0/casperjs.git /usr/local/casperjs && ln -sf /usr/local/casperjs/bin/casperjs /usr/local/bin/casperjs

CMD ["/usr/local/bin/casperjs", "test", "/test/*.js"]
```

```
docker run -ti -v `pwd`:/test example/casperjs casperjs test --output-encoding=utf8 --web-security=false --ssl-protocol=any --ignore-ssl-errors=true --verbose=true --logLevel=debug /test/test.js
```

```
casper.options.logLevel = 'debug';
casper.options.verbose = true;
casper.options.sslProtocol = 'any';
casper.options.viewportSize = {};
casper.options.viewportSize.width = 1280;
casper.options.viewportSize.height = 720;
casper.options.pageSettings.loadImages = false;
casper.options.pageSettings.loadPlugins = false;
casper.options.pageSettings.localToRemoteUrlAccessEnabled = true;
casper.options.pageSettings.webSecurity = false;
casper.options.pageSettings.userAgent = 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/39.0.2171.71 Safari/537.36';

casper.on("resource.error", function(resourceError){
    console.log('Unable to load resource (#' + resourceError.id + 'URL:' + resourceError.url + ')');
    console.log('Error code: ' + resourceError.errorCode + '. Description: ' + resourceError.errorString);
});

casper.on('resource.requested', function(request){
    console.log('Resource requested');
    console.log(JSON.stringify(request));
});

casper.on('resource.received', function(resource) {
    console.log('Resource received');
    console.log(JSON.stringify(resource));
});

casper.on("page.error", function(msg, trace) {
    console.log("Error: " + msg, "ERROR");
});

casper.on('waitFor.timeout', function (timeout, details) {
    casper.captureSelector('/test/waitForTimeout.png', 'selector');
    console.log(casper.getHTML('.importService'));
    console.log(JSON.stringify(details));
});

casper.on('remote.message', function(msg) {
    casper.log('remote message caught: ' + msg);
});

//http://stackoverflow.com/a/25359714
casper.on( 'page.initialized', function(){
    this.evaluate(function(){
        var isFunction = function(o) {
            return typeof o == 'function';
        };

        var bind,
            slice = [].slice,
            proto = Function.prototype,
            featureMap;

        featureMap = {
            'function-bind': 'bind'
        };

        function has(feature) {
            var prop = featureMap[feature];
            return isFunction(proto[prop]);
        }

        // check for missing features
        if (!has('function-bind')) {
            // adapted from Mozilla Developer Network example at
            // https://developer.mozilla.org/en/JavaScript/Reference/Global_Objects/Function/bind
            bind = function bind(obj) {
                var args = slice.call(arguments, 1),
                    self = this,
                    nop = function() {
                    },
                    bound = function() {
                        return self.apply(this instanceof nop ? this : (obj || {}), args.concat(slice.call(arguments)));
                    };
                nop.prototype = this.prototype || {}; // Firefox cries sometimes if prototype is undefined
                bound.prototype = new nop();
                return bound;
            };
            proto.bind = bind;
        }
    });
});

var delay = 10000;
casper
    .test
    .begin('tests', function suite(test) {
        casper
            .start()
            .open('url');

        casper.page.onFilePicker = function(oldFile) {***};

       some navigation/assert is done here


        casper.run(function() {
            test.done();
        });

});
```

**Comment by ariya** [comment]
Can you try it again with PhantomJS 2.1?

**Comment by jsmirnov** [comment]
@ariya 
Week ago, I used this Docker image: https://hub.docker.com/r/wernight/phantomjs/, which basically was using latest phantomjs master and it worked. So at least in week-ago-master it was already fixed.

While if week-ago-master is not included into 2.1 of course I can test it out.
